src/Language/PureScript/CodeGen/JS/Common.hs

```diff
@@ -75,107 +81,160 @@ identCharToString c = '$' : show (ord c)
+-- Checks whether a name matches a built-in value in Javascript.
 --
-properNameIsJsReserved :: String -> Bool
-properNameIsJsReserved name =
-  name `elem` [ "Infinity"
-              , "NaN"
-              , "Object"
-              , "Function"
-              , "Boolean"
-              , "Error"
-              , "EvalError"
-              , "InternalError"
-              , "RangeError"
-              , "ReferenceError"
-              , "SyntaxError"
-              , "TypeError"
-              , "URIError"
-              , "Number"
-              , "Math"
-              , "Date"
-              , "String"
-              , "RegExp"
-              , "Array"
-              , "Int8Array"
-              , "Uint8Array"
-              , "Uint8ClampedArray"
-              , "Int16Array"
-              , "Uint16Array"
-              , "Int32Array"
-              , "Uint32Array"
-              , "Float32Array"
-              , "Float64Array"
-              , "ArrayBuffer"
-              , "DataView"
-              , "JSON"
-              , "Intl" ]
+nameIsJsBuiltIn :: String -> Bool
+nameIsJsBuiltIn name =
+  elem name
+    [ "Infinity"
+    , "NaN"
+    , "undefined"
+    , "eval"
+    , "uneval"
+    , "isFinite"
+    , "isNaN"
+    , "parseFloat"
+    , "parseInt"
+    , "decodeURI"
+    , "decodeURIComponent"
+    , "encodeURI"
+    , "encodeURIComponent"
+    , "escape"
+    , "unescape"
+    , "Object"
+    , "Function"
+    , "Boolean"
+    , "Symbol"
+    , "Error"
+    , "EvalError"
+    , "InternalError"
+    , "RangeError"
+    , "ReferenceError"
+    , "SyntaxError"
+    , "TypeError"
+    , "URIError"
+    , "Number"
+    , "Math"
+    , "Date"
+    , "String"
+    , "RegExp"
+    , "Array"
+    , "Int8Array"
+    , "Uint8Array"
+    , "Uint8ClampedArray"
+    , "Int16Array"
+    , "Uint16Array"
+    , "Int32Array"
+    , "Uint32Array"
+    , "Float32Array"
+    , "Float64Array"
+    , "Map"
+    , "Set"
+    , "WeakMap"
+    , "WeakSet"
+    , "SIMD"
+    , "ArrayBuffer"
+    , "DataView"
+    , "JSON"
+    , "Promise"
+    , "Generator"
+    , "GeneratorFunction"
+    , "Reflect"
+    , "Proxy"
+    , "Intl"
+    , "arguments"
+    ]
+
+jsAnyReserved :: [String]
+jsAnyReserved =
+  concat
+    [ jsReserved
+    , jsFutureReserved
+    , jsFutureReservedStrict
+    , jsOldReserved
+    , jsLiterals
+    ]
+
+jsReserved :: [String]
+jsReserved =
+  [ "break"
+  , "case"
+  , "catch"
+  , "class"
+  , "const"
+  , "continue"
+  , "debugger"
+  , "default"
+  , "delete"
+  , "do"
+  , "else"
+  , "export"
+  , "extends"
+  , "finally"
+  , "for"
+  , "function"
+  , "if"
+  , "import"
+  , "in"
+  , "instanceof"
+  , "let"
+  , "new"
+  , "return"
+  , "super"
+  , "switch"
+  , "this"
+  , "throw"
+  , "try"
+  , "typeof"
+  , "var"
+  , "void"
+  , "while"
+  , "with"
+  , "yield"
+  ]
+
+jsFutureReserved :: [String]
+jsFutureReserved =
+  [ "enum"
+  , "await"
+  ]
+
+jsFutureReservedStrict :: [String]
+jsFutureReservedStrict =
```

Comment:
I found `static` [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Lexical_grammar#Future_reserved_keywords), is that incorrect then?